### PR TITLE
feat: upgrade to regex

### PR DESCRIPTION
**Pull request by TGTGamer**
Heya pal,

I really like the concept of this. I'm trying to move away from using bots within my GitHub projects and towards actions. You can see my (painfully long) workflow template [here](https://github.com/Videndum/Universal-GitAction-Workflows) 

So I've got a few improvements I'd love to suggest and I've even thrown in a simple PR for them. Check out the comments to help you figure out what happened xD

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+     * @since 1.0.0
+     */
+    const debtIssueTitle = await parseContent(
+      prBody,
+      options.titleStart,
+      options.titleEnd,
+      options.bodyEndRegex
+    ).catch(err => {
+      core.setFailed(err)
+    })
```
Comment (TGTGamer) [inline src/index.ts:54]: Converted this section is now a promise

`core.getInput('body_start_regex')`
https://github.com/teepsdonnelly/debt-collector/blob/bace221a66193572383fcf9c3e740966ebcd93f3/index.js#L10-L11

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    /**
+     * Finds the regex start point and gets the index number of it
+     * TODO: Cleanup because this ugly xD
+     * @author TGTGamer
+     * @since 2.0.0
+     */
+    const contentStart = body.match(`/${StartRegex}/im`)
+    if (!contentStart || !contentStart.index)
+      throw new Error('Start not matched')
+    const contentStartIndex = contentStart.index + contentStart.length
```
Comment (TGTGamer) [inline src/index.ts:129]: Section now uses Regex to find the index of the descriptor

https://github.com/teepsdonnelly/debt-collector/blob/bace221a66193572383fcf9c3e740966ebcd93f3/index.js#L29-L35

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+     * @since 2.0.0
+     */
+    const contentEnd = body.match(EndRegex)
+    if (!contentEnd || !contentEnd.index) throw new Error('End not matched')
+    var contentEndIndex: number
+    if (endMode == 'regex') {
+      contentEndIndex = contentEnd.index
+    } else {
+      contentEndIndex = contentEnd.index + contentEnd.length
+    }
```
Comment (TGTGamer) [inline src/index.ts:143]: Section now uses regex to close a section. 

https://github.com/teepsdonnelly/debt-collector/blob/bace221a66193572383fcf9c3e740966ebcd93f3/index.js#L35-L43

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    body:
+      issueBody +
+      spacer +
+      'See the [Pull Request that created this Issue](' +
+      prHtmlURL +
+      ')'
+  })
+
+  console.log('Issue created: ' + titlePrefix + ' ' + issueTitle)
+}
```
Comment (TGTGamer) [inline src/index.ts:188]: User can now define the prefix

https://github.com/teepsdonnelly/debt-collector/blob/bace221a66193572383fcf9c3e740966ebcd93f3/index.js#L67

Not sure why, but you had a function inside a function? Was there a reason for this? 
https://github.com/teepsdonnelly/debt-collector/blob/bace221a66193572383fcf9c3e740966ebcd93f3/index.js#L54-L76

File: src/index.ts (hunk 1 above)
Comment (teepsdonnelly) [reply src/index.ts:54]: Love the idea of using regex for this!

File: src/index.ts (hunk 4 above)
Comment (teepsdonnelly) [reply src/index.ts:188]: I likely tried something else that didn't work, but it was more than likely due to my inexperience with Actions / JavaScript -- I've primarily been an iOS developer :)

**Pull request merged**